github-bot/harvester_github_bot/__init__.py

```diff
@@ -108,3 +113,95 @@ def issue_transfer(form):
             issue.create_comment('Automation e2e test issue: {}'.format(issue_test_link))
     else:
         app.logger.debug('label require/automation-e2e does not exists, not creating test issue')
+
+
+@app.route('/github', methods=['POST'])
+@auth.login_required
+def gh():
+    res = request.get_json()
+
+    msg = "Skip action"
+    if res.get('action') is not None and res['action'] == 'labeled' and res.get('issue') is not None:
+        msg = backport(res)
+
+    return msg, 200
+
+
+# link: https://github.com/harvester/harvester/issues/2324
+# Title: [Backport v1.x] copy-the-title.
+# Description: backport the issue #link-id
+# Copy assignees and all labels except the backport-needed and add the not-require/test-plan label.
+# Move the issue to the associated milestone and release.
+def backport(obj):
```

Comment:
can we refactor this file with:
1. moving the operating method to a separated file, keep the init file clean and tidy, e.g. the route and init methods only.
2. break down the big function e.g backport into reusable methods.